### DISCUSSION TITLE
Compose Manager : Download plugins are empty. How to use them?

### DISCUSSION THREAD

**Discussion by vippium** [ReVanced/revanced-discussions#1115] [Question]
How to use the download plugins in RV Manager Compose. There is a option for that but shows `No plugins installed`.

**Comment by optimatrix0** [comment]
I also have this same doubt

**Comment by optimatrix0** [comment]
probably the plugins have not been released yet

**Comment by Ushie** [comment]
ReVanced Manager Compose is in development and is not ready to be used, plugins have not yet been developed

**Comment by optimatrix0** [comment]
In my mind, I thought that the webview plugin would be the standard incorporated into the Manager, and people would have the opportunity to download others, but without any plugins to install, it becomes complicated for us to learn how to use it. Of course, I am thinking about new users, not those who already use it. I agree with some developers who said that a webview plugin for users to find and click the download button is ideal to be incorporated. So I believe that creating this type of plugin will solve both problems. The sincere opinion of a user who really likes the project is that they want to see the new version of the Manager soon.

**Reply by vippium** [reply]
+1 for this. This type of plugin is there in v2 compose, but removed later in v3. It should be there or can't dev's make it compatible v2 downloader plugin for v3 compose?

**Reply by oSumAtrIX** [reply]
ReVanced will offer first party downloader(s) in partnership with sites. APKMirror for example. We will not offer downloaders that abuse a service though. For example for APKMirror we will display a webview or other ways for them to generate revenue instead of abusing their API. The API for downloaders is published eventually so you can write your own downloaders for any site, irrespectively of us.

**Reply by vippium** [reply]
But atleast you can make compose to support or patch any version of apps. It only patches the recommended version. But if user want to patch higher or lower version (for deprecated features obviously), he can't. After downloading the apk from downloader, it throws error of recommended version. If you prefer, then, creating a toggle for this which will allow user whether he wants to always patch with recommended version or to patch with other versions.

I Hope you'll understand what I want to say.

**Reply by oSumAtrIX** [reply]
The downloader has nothing to do with deciding which version to patch, that's specified by the patches and for that a toggle already exists

**Reply by vippium** [reply]
> The downloader has nothing to do with deciding which version to patch, that's specified by the patches and for that a toggle already exists

You are still not understanding what I'm saying.
Check this ss:
![Screenshot_2025_0109_223029](https://github.com/user-attachments/assets/2db9ed49-c5e5-4747-8dbc-5a7ec318e94f)

This shows that downloader completely depends on patches recommended version.
If required, i'll give you the recording of what I want to say,

**Reply by oSumAtrIX** [reply]
That is unrelated as the downloader requests version X but version Y is returned by the site.

**Comment by optimatrix0** [comment]
Are you expecting users to manually import a previously downloaded APK if it's not supported? This contradicts the Manager's purpose of simplifying app correction for Android users.

My final question: What's the primary reason for migrating from Flutter to Jetpack in the Manager? Clarifying this would benefit many users, not just me, as the repository explanation remains superficial.

**Reply by oSumAtrIX** [reply]
> Are you expecting users to manually import a previously downloaded APK if it's not supported? This contradicts the Manager's purpose of simplifying app correction for Android users.

Support for a version is determined by the patches. Apps can be selected manually or downloaded via downloaders, as a simplification.

> My final question

Flutter is clunky and Jetpack compose easier to use

**Comment by optimatrix0** [comment]
I appreciate your previous responses. My doubts about migrating from Flutter to Jetpack have been clarified.

I'd like to highlight a concern regarding Manager plugins. Allowing external downloads and updates within Manager is crucial. This ensures Manager fulfills its primary goal: simplifying Android app downloads and corrections.

Looking forward to the project's completion! You're doing great work!

Thanks!

**Reply by vippium** [reply]
Ya..If users still have to download old version to use them, then what's the objective of using downloader? Instead of using APK Mirror API (obviously for revenue purpose), just add another plugin with uptodown api (or any other) to support old version download also.

**Reply by oSumAtrIX** [reply]
Downloaders aren't restricted to downloading one version.

**Comment by Axelen123** [comment]
The reasons for requiring the user to select the recommended version are explained in [this issue](https://github.com/ReVanced/revanced-manager/issues/2358#issuecomment-2558657370). An APKMirror plugin is available [here](https://github.com/ReVanced/revanced-manager-apkmirror-downloader). The README explains how to use it.
